src/theory/strings/theory_strings_rewriter.cpp

```diff
@@ -1691,44 +1692,23 @@ RewriteResponse TheoryStringsRewriter::postRewrite(TNode node) {
   }
   else if (nk == kind::STRING_ITOS)
   {
-    if(node[0].isConst()) {
-      if( node[0].getConst<Rational>().sgn()==-1 ){
-        retNode = NodeManager::currentNM()->mkConst( ::CVC4::String("") );
-      }else{
-        std::string stmp = node[0].getConst<Rational>().getNumerator().toString();
-        Assert(stmp[0] != '-');
-        retNode = NodeManager::currentNM()->mkConst( ::CVC4::String(stmp) );
-      }
-    }
+    retNode = RewriterStr::rewriteIntToStr(node);
   }
   else if (nk == kind::STRING_STOI)
   {
-    if(node[0].isConst()) {
-      CVC4::String s = node[0].getConst<String>();
-      if(s.isNumber()) {
-        retNode = nm->mkConst(s.toNumber());
-      } else {
-        retNode = nm->mkConst(Rational(-1));
-      }
-    } else if(node[0].getKind() == kind::STRING_CONCAT) {
-      for(unsigned i=0; i<node[0].getNumChildren(); ++i) {
-        if(node[0][i].isConst()) {
-          CVC4::String t = node[0][i].getConst<String>();
-          if(!t.isNumber()) {
-            retNode = NodeManager::currentNM()->mkConst(::CVC4::Rational(-1));
-            break;
-          }
-        }
-      }
-    }
+    retNode = RewriterStr::rewriteStrToInt(node);
   }
   else if (nk == kind::STRING_IN_REGEXP)
   {
     retNode = rewriteMembership(node);
   }
   else if (nk == STRING_TO_CODE)
   {
-    retNode = rewriteStringToCode(node);
+    retNode = RewriterStr::rewriteStringToCode(node);
+  }
+  else if (nk == STRING_FROM_CODE)
+  {
+    retNode = RewriterStr::rewriteStringFromCode(node);
```

Comment:
Instead of having those here, the object-oriented way to do this would be to override `postRewrite` in `RewriterStr` and then to call the original `postRewrite` and add those cases in addition to that (otherwise the inheritance on `RewriterStr` doesn't seem to get us much).

Reply:
This actually isn't possible currently since static functions can't be virtual.  (Is this changing in the future?)